### PR TITLE
Fix/board member service

### PR DESCRIPTION
**Pull request by Chanuth10**
Fix the bugs in BoardMember Service

### CODE REVIEW

**Review by rusiruavb — APPROVED** [review]
Changes are LGTM. For update, there are some other solutions as well. But for now this is enough. When we find new solution to update document, we will apply to the codebase.

File: src/api/services/BoardMember.service.ts [hunk 1]
```diff
@@ -46,24 +46,43 @@ export const getAllBoardMembers = async () => {
...
  * @param updateData @type DocumentDefinition<IBoardMember>
  */
+
 export const updateBoardMemberDetails = async (
   boardMemberId: string,
   updateData: DocumentDefinition<IBoardMember>
 ) => {
   return await BoardMemberModel.findById(boardMemberId)
     .then(async (boardMemberDetails) => {
       if (boardMemberDetails) {
```
Comment (senuravihanjayadeva) [inline src/api/services/BoardMember.service.ts:56]: this is not needed. Please remove this if condition

File: src/api/services/BoardMember.service.ts [hunk 2]
```diff
@@ -46,24 +46,43 @@ export const getAllBoardMembers = async () => {
...
-        boardMemberDetails.socialMedia.facebook =
-          updateData.socialMedia.facebook;
-        boardMemberDetails.socialMedia.instagram =
-          updateData.socialMedia.instagram;
-        boardMemberDetails.socialMedia.linkedIn =
-          updateData.socialMedia.linkedIn;
-        boardMemberDetails.socialMedia.twitter = updateData.socialMedia.twitter;
-        return await boardMemberDetails.save();
+        if (boardMemberDetails.deletedAt === null) {
+          if (updateData.name !== "") {
```
Comment (senuravihanjayadeva) [inline src/api/services/BoardMember.service.ts:58]: if (updateData.name) this is enogh. Do the same for other if conditions

File: src/api/services/BoardMember.service.ts [hunk 3]
```diff
@@ -46,24 +46,43 @@ export const getAllBoardMembers = async () => {
...
+            boardMemberDetails.name = updateData.name;
+          }
+          if (updateData.position !== "") {
+            boardMemberDetails.position = updateData.position;
+          }
+          if (updateData.image !== "") {
+            boardMemberDetails.image = updateData.image;
+          }
+          if (updateData.socialMedia.facebook !== "") {
```
Comment (senuravihanjayadeva) [inline src/api/services/BoardMember.service.ts:67]: use below if condition

if(updateData.socialMedia){

}

**Review by senuravihanjayadeva — CHANGES_REQUESTED** [review]

**Review by senuravihanjayadeva — APPROVED** [review]

**Pull request merged**